### PR TITLE
Enhance CI settings (avoid the need to manually add new apps)

### PR DESCRIPTION
**Pull request by cuducos**
Right now, from the list of #24, I'm just addressing the _minimize need for manual maintenance_ (all new modules are included by default in Coveralls and Code Climate), which IMHO was the most important thing.

I still can go further with:

* Using some defaults for some settings and minimize the usage of the web panel for _env vars_;
* If so, I'd probably use [dj-database-url](https://github.com/kennethreitz/dj-database-url);
* And surely investigate what is causing timeout in Travis, i.e. #21

Any thoughts?

### CODE REVIEW

**Comment by cuducos** [comment]
@dirtycoder I'm gonna push a new commit that will fail on Travis CI until we update the environment variables needed.

You can delete all of them except `OPBEAT_DISABLE_SEND`, and add `DATABASE_URL=postgres://postgres@localhost/pets`, ok?

When you have only these vars, can you restart the build at Travis?

File: pets/pets/settings/prod.py [hunk 1]
```diff
@@ -112,16 +113,9 @@
...
-    'default': {
-        'ENGINE': config('DB_ENGINE'),
-        'NAME': config('DB_NAME'),
-        'USER': config('DB_USERNAME'),
-        'PASSWORD': config('DB_PASSWORD'),
-        'HOST': config('DB_IP'),
-        'PORT': config('DB_PORT'),
-        'CONN_MAX_AGE': config('DB_CONN_MAX_AGE', cast=int),
-    }
+    'default': config('DATABASE_URL', default=default_db, cast=db_url)
```
Comment (dirtycoder) [inline pets/pets/settings/prod.py]: How can we add the CONN_MAX_AGE setting here?

**Comment by dirtycoder** [comment]
Reading the code I just spotted an error I introduced some time ago, [here](https://github.com/dirtycoder/pets/pull/26/files#diff-8b4380597f4115da0444e0b08573e361R22).
`Csv()` expects a string, so the correct value would be an empty string `''`, but I have another idea.

For the `SECRET_KEY` and the `DATABASE_URL` I'd prefer to not have a default value, as it's an important configuration I'd want it to be a complete "show stopper" if not properly configured, instead of letting the user run the project with a wrong setting.
Running with the default could lead to bad results like the file system being erased, in case of sqlite and Heroku, or creating a security problem running with the known `secret_dummy_key` value for `SECRET_KEY`.

**Comment by dirtycoder** [comment]
Reading the code I just spotted an error I introduced some time ago, [here](https://github.com/dirtycoder/pets/blob/next/pets/pets/settings/prod.py#L21).
`Csv()` expects a string, so the correct value would be an empty string `''`.

For the `SECRET_KEY` and the `DATABASE_URL` I'd prefer to not have a default value, as it's an important configuration I'd want it to be a complete "show stopper" if not properly configured, instead of letting the user run the project with a wrong setting.
Running with the default could lead to bad results like the file system being erased, in case of sqlite and Heroku, or creating a security problem running with the known `secret_dummy_key` value for `SECRET_KEY`.

File: .coveragerc [hunk 2]
```diff
@@ -0,0 +1,13 @@
+[run]
+
+# include everything inside pets/pets/ by default
+source=.
+
+# exlcude tests, migrations, settings etc.
```
Comment (dirtycoder) [inline .coveragerc]: Typo

File: README.md [hunk 3]
```diff
@@ -52,13 +52,12 @@ Then you have to set following variables:
 
 #### Database
 
-* **`DB_NAME`**, **`DB_USERNAME`**, **`DB_PASSWORD`**, **`DB_IP`**, **`DB_PORT`**: Credentials to access Pets's database.
-* **`DB_ENGINE`**: Django databse backend to connect to Pets's database (e.g. `django.db.backends.postgresql_psycopg2`)
-* **`DB_CONN_MAX_AGE`**: (e.g. `0`) [Django's database `CONN_MAX_AGE`](https://docs.djangoproject.com/en/1.8/ref/settings/#std:setting-CONN_MAX_AGE`)
```
Comment (dirtycoder) [inline README.md]: We should not remove `DB_CONN_MAX_AGE` here.

File: pets/pets/settings/prod.py (hunk 1 above)
Comment (cuducos) [inline pets/pets/settings/prod.py]: Sure thing!

**Comment by cuducos** [comment]
Great ideas, man!
I basically addressed all the issues you raised in the scattered commits above. Does it look better?

**Comment by dirtycoder** [comment]
@cuducos sorry for the slow pace of review, a lot of things to do.
Anyway, everything looks great now, thanks!
I think we are good to go here, could you rebase this in a single commit? :)

The last build timed out, don't know if you can get the information needed to investigate the problem from the log.
But here is an idea, I think you could configure your own repo on Travis CI and then ask them to enable de debug there, so you could connect to the build machine.

**Comment by cuducos** [comment]
> I think we are good to go here, could you rebase this in a single commit? :)

Great news, then!

I still have to investigate the timeout thing (not only this one, but #21 too) before rebasing and considering it a _ready to merge_ PR hahaha…

**Comment by dirtycoder** [comment]
As #21 is the issue that really deals with the timeout, I'm ok to merging this one, if the tests passes when a timeout don't occur.

I think the current PR deals with the issues raised in #24, but with the rebase and a new commit push, the tests will run again and a new timeout could not occur.

Thank you very much.

I'll leave this up to you to decide what to do. :)

**Comment by cuducos** [comment]
Already running tests on my fork, allow me a few days of exploration, if I figure out anything we fix here. Otherwise we keep #21 alive and merge this PR, ok? I'll come back with some news soon!

**Comment by dirtycoder** [comment]
Ok, thanks! :)
Great!

**Comment by cuducos** [comment]
I got in touch with Travis support and in fact the containers, Selenium and PhantomJS aren't best friends. There is [an approach of _retries_](https://github.com/gator-life/gator.life/commit/79f4e5e584e7f691d954a67e8e54f236abb91e36) that seem to be working in other projects, but I don't have time to implement this on Pets that quickly.

So it's up to you: feel free to merge this and leave #21 for another PR (let me know and I paste this reference there), or keep this PR open until one of us are able to try to implement this workaround.

**Comment by dirtycoder** [comment]
@cuducos hey man thx for you time!
I'll merge this PR an close the related issue once the Travis builds are green. :tada: 

I'll leave the #21 open for now, I was thinking about it and probably I'll migrate to Firefox + xvfb in the CI server as it feels less "hacky" and people writing test code don't need to have an extra care using functions that are not standard.

**Pull request merged**